src/runtime/RegExpObject.h

```diff
@@ -168,8 +169,9 @@ class RegExpObject : public Object {
     Option m_option;
     JSC::Yarr::YarrPattern* m_yarrPattern;
     JSC::Yarr::BytecodePattern* m_bytecodePattern;
-
     SmallValue m_lastIndex;
+    NamedCapturesGroups m_namedCaptureGroups;
```

Comment:
It seems that `m_namedCaptureGroups` could be replaced by the existing `YarrPattern::m_captureGroupNames`. Would you please check it?

Reply:
In my first draft i used the `YarrPattern::m_captureGroupNames`, which essentially stores the same data, but it is not created with `GCUtil::gc_malloc_allocator` in mind. This leads to an issue, if we run more than a dozen tests, then the structure gets cleared out and segfaults happens.

Reply:
Could you give me a sample test code that `YarrPattern::m_captureGroupNames` gets cleared? I'll check it. Thank you.

Reply:
I created a new branch which use `YarrPattern::m_captureGroupNames`, instead of `NamedCapturesGroups` and also added a test file namely `Segmentation-fault-naming-group.js`.

https://github.com/kisbg/escargot/tree/newfeature-naminggroupTEST

i hope it's going to be any help! :)

Reply:
I have uploaded a patch (https://github.com/Samsung/escargot/pull/563) which allocates `YarrPattern::m_captureGroupNames` on GC heap. Please rebase this pr and use `YarrPattern::m_captureGroupNames` instead of `RegExpObject::m_namedCaptureGroups` because adding this member to all RegExpObject looks inefficient.